### PR TITLE
Add rake task to validate schemas

### PR DESCRIPTION
**Pull request by jhirbour**
Add validation checks for the schemas that are used. This will allow better tracking of when the un-published APIs change.

### CODE REVIEW

File: lib/ncua/client_validator.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    def self.list_fields_valid?
+      expected_keys = ['CU_NAME', 'AddressLongitude', 'AddressLatitude',
+                       'CU_SITENAME', 'CU_NUMBER', 'City', 'CityPhysical',
+                       'Country', 'index', 'IsMainOffice', 'Phone',
+                       'SiteFunctions', 'SiteId', 'SiteName', 'State', 'URL',
+                       'Zipcode', 'distance', 'Street'].sort
+      found_keys = Client.new.find_credit_union_by_charter_number(42)["list"].first.keys.sort
+      # return expected_fields is a subset of found_keys
+      return found_keys == expected_keys
```
Comment (treznick) [inline lib/ncua/client_validator.rb]: D'oh! We'll fix this

File: lib/ncua/credit_union/scraper_validator.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      def has_a_table?
+        !html_doc.at_css('table#MainContent_newDetails').nil?
+      end
+
+      def has_correctly_formatted_key_value_rows?
+        value = true
+        key_value_rows.each do |tr|
+          tr_cells = tr.css("td")
+
+          value = (key_value_row?(tr_cells) || sub_header_row?(tr_cells))
```
Comment (treznick) [inline lib/ncua/credit_union/scraper_validator.rb]: this could turn into something like `.select { |tr| OR_CONDITION }.length > 0`

**Comment by danbernier** [comment]
I :heart: the explanation in the README. :+1: 

Rather than store the rake task in a .rakefile, and make users have to go that crazy `import "#{Gem::Specification.find_by_name('ncua').gem_dir}/lib/tasks/ncua.rake"` thing, you can just make a `Rake::Task` subclass in a `.rb` file, and they can `require` it when they need it. If you look at your Rakefile, you'll see examples there already: `RSpec::Core::RakeTask.new(:spec)`

File: lib/ncua/credit_union/scraper_validator.rb (hunk 2 above)
Comment (danbernier) [inline lib/ncua/credit_union/scraper_validator.rb]: Or even better: 

```
key_value_rows.all? { |tr|
  ...
}
```

**Comment by treznick** [comment]
Done and Done! good to :ship: :sheep: ?

**Comment by danbernier** [comment]
Fine by me
On Nov 4, 2015 4:31 PM, "Tom Reznick" notifications@github.com wrote:

> Done and Done! good to [image: :ship:] [image: :sheep:] ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ContinuityControl/ncua/pull/4#issuecomment-153871552.

## 

## 

---

This email and all files transmitted with it are intended solely for use by 
the intended recipient.  Any other use is prohibited.  If you received this 
email in error, please notify the sender, delete the email and destroy all 
attachments.  Opinions and ideas expressed herein are solely those of the 
sender, and may not reflect the official position of Continuity.

**Pull request merged**